uaclient/util.py

```diff
@@ -497,6 +497,54 @@ def configure_web_proxy(
         request.install_opener(opener)
 
 
+def configure_snap_proxy(
+    http_proxy: "Optional[str]", https_proxy: "Optional[str]"
+) -> None:
+    """
+    Configure snap to use http and https proxies.
+
+    :param http_proxy: http proxy to be used by snap. If None, it will
+                       not be configured
+    :param https_proxy: https proxy to be used by snap. If None, it will
+                        not be configured
+    """
+    if http_proxy:
+        subp(["snap", "set", "system", "proxy.http={}".format(http_proxy)])
+
+    if https_proxy:
```

Comment:
I think this function also needs to support the following:
  1. checking existing snap config values, if there are existing proxy settings != to those requested,  prompt with the following message before changing values:
  ```bash
  Setting the https_proxy will overwrite the following system settings:
  snap config system proxy.https = https://blah
  canonical-livepatch = https//livepatch-proxy
Are you sure (Y/n):
```
  2.  Support unsetting snap proxy config values when https_proxy="" or None
  3. Support for unsetting snap http_proxy value if http_proxy == "" or None

Reply:
1. @orndorffgrant is working on that promp functionality on apt proxies. I think the logic can be generic enough to be used here.

2 and 3. Completely agree

Reply:
+1 the prompt can be a separate PR if needed. no need to block this PR on that.

Reply:
So I don't see any code that does (2) or (3). That is, we aren't calling `snap unset ...` anywhere when our config doesn't have a proxy value. I think that this current state is good. We can't safely remove a proxy config in snap/livepatch because of the following:

If our config does not have a `http_proxy` value, and snap is not configured with a proxy, then there is nothing to do. Calling `snap unset ...` won't harm anything though. This is not a problem case - the following is:

If our config does not have a `http_proxy` value, but snap is _already configured_ to use a proxy, then we could be in one of two scenarios:
1. the user previously _used UA_ to set a proxy, then _used UA_ to unset the proxy. In this scenario, we want to run `snap unset...`
2. the user previously _manually ran_ `snap set...` to set the proxy, and hasn't touched the UA config at all. In this scenario, we shouldn't `snap unset...`. It would also be confusing if `ua enable livepatch` prompted to remove the user's snap proxy (even worse: if we went with a prompt in this scenario then a `ua enable livepatch --assume-yes` would unexpectedly remove the users snap proxy).

The problem is: Given a uaclient.conf with no proxy set and seeing that snap is already configured to use a proxy, we don't know if we're in scenario 1 or 2.

We will be able to tell which scenario we're in if the user runs `ua config unset http_proxy`, so in that case, we can safely run `snap unset...`, but otherwise, I don't think we can.

All that is to say: I think this PR does not need `snap unset...` or the `canonical-livepatch` equivalent, which means that it is good as is.

> As an aside: this complication may prevent the prompting logic from being nicely shared between the apt proxy and snap/livepatch proxy configuration... we'll see as I work through it.

Reply:
That's an excellent point here. I was kind of performing that by assuming that if `http_proxy=""` then we will unset it. Even though I don't call unset directly, it seems that setting the proxy as empty produces the same expected output on both livepatch and snap.

But this is really problematic as you pointed in scenario 2. Specially if we start delivering `uaclient.conf` with `ua_config` like this:

```
ua_config:
  http_proxy: ""
  https_proxy: ""
```

Maybe the safest way for us to handle that is to just unset proxies on the `ua config unset` scenario. But let's bring this with the team and better discuss it.

Reply:
Ah and good point about the empty string values! I wasn't even thinking about that. Yes lets discuss this more today